boa/src/exec/mod.rs

```diff
@@ -68,18 +68,18 @@ impl Executor for Interpreter {
     }
 
     #[allow(clippy::match_same_arms)]
-    fn run(&mut self, expr: &Expr) -> ResultValue {
-        match expr.def {
-            ExprDef::Const(Const::Null) => Ok(to_value(None::<()>)),
-            ExprDef::Const(Const::Undefined) => Ok(Gc::new(ValueData::Undefined)),
-            ExprDef::Const(Const::Num(num)) => Ok(to_value(num)),
-            ExprDef::Const(Const::Int(num)) => Ok(to_value(num)),
+    fn run(&mut self, expr: &Node) -> ResultValue {
+        match *expr {
```

Comment:
Changed from expr to node.
```suggestion
    fn run(&mut self, node: &Node) -> ResultValue {
        match *node {
```